### PR TITLE
docs: Add initial docs for the Meson CI [skip ci]

### PR DESCRIPTION
**Pull request by mensinda**
This is not complete yet (Azure is missing for instance), but this should at least explain how the docker images are created.

### CODE REVIEW

File: docs/markdown/snippets/MesonCI.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Meson CI setup
+
+This document is aimed for Meson contributors and documents
+the CI setup used for testing Meson itself. The Meson
+project uses multiple CI platforms for covering a wide
+range of target systems.
+
+## Travis CI
+
+The travis configuration file is the `.travis.yml` int the
```
Comment (jon-turney) [inline docs/markdown/snippets/MesonCI.md:10]: int -> in

File: docs/markdown/snippets/MesonCI.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+### Image generation
+
+There are no manual Dockerfiles. Instead the Dockerfile is
+automatically generated by the `build.py` script. This is
+done to ensure that all images have the same layout and can
+all be built and tested automatically.
+
+The Dockerfile is generated from the `image.json` file and
+basically only adds a few common files and runns the
```
Comment (jon-turney) [inline docs/markdown/snippets/MesonCI.md:41]: runns -> runs

File: docs/markdown/snippets/MesonCI.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+`install.sh` script which should contain all distribution
+specific setup stepps. The `common.sh` can be sourced via
```
Comment (jon-turney) [inline docs/markdown/snippets/MesonCI.md:43]: stepps -> steps

File: docs/markdown/snippets/MesonCI.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+This document is aimed for Meson contributors and documents
+the CI setup used for testing Meson itself. The Meson
+project uses multiple CI platforms for covering a wide
+range of target systems.
+
+## Travis CI
+
+The travis configuration file is the `.travis.yml` int the
+the project root. This platform tests cross compilation and
+unit builds on a [linux docker image](#docker-images) and
```
Comment (jon-turney) [inline docs/markdown/snippets/MesonCI.md:12]: What is a 'unit build'?  unity? 'unit tests'?

File: docs/markdown/snippets/MesonCI.md (hunk 1 above)
Comment (mensinda) [reply docs/markdown/snippets/MesonCI.md]: Fixed, thanks.

**Comment by scivision** [comment]
I see MSYS2 has been having issues starting. Not sure if this would help, using GitHub Actions for MSYS2:

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">We now have a GitHub Action for easily setting up a CI job that uses MSYS2. Gets things up and running with a few lines of yaml -&gt; <a href="https://t.co/GmK5bykn13">https://t.co/GmK5bykn13</a> <a href="https://t.co/thaLQ6hnNb">pic.twitter.com/thaLQ6hnNb</a></p>&mdash; msys2org (@msys2org) <a href="https://twitter.com/msys2org/status/1275125348024401922?ref_src=twsrc%5Etfw">June 22, 2020</a></blockquote>

**Pull request merged**